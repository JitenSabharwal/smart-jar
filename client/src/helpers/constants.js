export const ADD_CONTAINER_REQUEST = 'ADD_CONTAINER_REQUEST'
export const ADD_CONTAINER_SUCCESS = 'ADD_CONTAINER_SUCCESS'
export const ADD_CONTAINER_FAILURE = 'ADD_CONTAINER_FAILURE'
export const UPDATE_CONTAINER_REQUEST = 'UPDATE_CONTAINER_REQUEST'
export const UPDATE_CONTAINER_SUCCESS = 'UPDATE_CONTAINER_SUCCESS'
export const UPDATE_CONTAINER_FAILURE = 'UPDATE_CONTAINER_FAILURE'
export const DELETE_CONTAINER = 'DELETE_CONTAINER'
export const DELETE_ALL_CONTAINERS = 'DELETE_ALL_CONTAINERS'
export const SELECT_CONTAINER = 'SELECT_CONTAINER'
export const CONTAINER_TABLE_HEADS = [
  { id: 'firstName', numeric: false, disablePadding: true, label: 'FirstName' },
  { id: 'lastName', numeric: false, disablePadding: false, label: 'LastName' },
  { id: 'hobbies', numeric: false, disablePadding: false, label: 'Hobbies' },
  { id: 'birthDate', numeric: false, disablePadding: false, label: 'BirthDate' },
  { id: 'url', numeric: false, disablePadding: false, label: 'Profile' },
]
export const DEFAULT_URL = '/uploads/default.jpeg'
export const MAX_LENGTH = 30
export const NAME_REGEX = /^[a-zA-Z ]{2,30}$/
