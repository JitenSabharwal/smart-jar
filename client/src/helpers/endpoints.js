export const ALL_CONTAINER_ENDPOINT = '/api/v1/containers'
export const ADD_CONTAINER_ENDPOINT = '/api/v1/containers/addContainer'
export const DELETE_CONTAINER_ENDPOINT = '/api/v1/containers/removeContainer'
export const DELETE_ALL_CONTAINER_ENDPOINT = '/api/v1/containers/removeAllContainers'
