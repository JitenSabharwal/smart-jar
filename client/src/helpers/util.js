import axios from 'axios'

/**
 * Function to do a prepare request headers for anything required
 */
export const perpareHeader = () => {
  const id = localStorage.getItem('id_token')
  const config = {
    headers: {
      'Authorization': id ? `Bearer ${id}` : '',
    },
  }
  return config
}

/**
 * Function to do a post Request
 * @param {string} url
 * @param {object} payload
 */
export const postRequest = (url, payload) => {
  const headers = perpareHeader()
  return axios.post(url, payload, headers)
}

/**
 * Function to do a get Request
 * @param {string} url
 * @param {object} payload
 */
export const getRequest = (url, payload) => {
  const headers = perpareHeader()
  return axios.get(url, headers)
}
/**
 * Funtion to filter data out
 * @param {array} data [{id: 'redquired', ...container properties}]
 * @param {*} val
 * @param {string} key If not _id any other prop
 */
export const findUser = (data, val, key = '_id') => {
  return data.filter(u => u[key] === val)
}

/**
 * Functiom to load set an initial state
 */
export const loadInitailState = (url) => {
  return new Promise((resolve, reject) => {
    getRequest(url)
      .then(resp => {
        const {success, data} = resp.data
        if (!success) {
          // If there was a problem, we want to
          return reject(resp)
        } else {
          // If login was successful, set the token in local storage
          const state = {
            container: {
              list: data || [],
            },
          }
          resolve(state)
        }
      })
      .catch(err => {
        reject(new Error('Error: ', err))
      })
  })
}

/**
 * Function returns -1, 0, 1
 *  depending on the match of the
 *  property orderBy
 *  of elements a and b
 * @param {object} a
 * @param {object} b
 * @param {string} orderBy
 */
function desc (a, b, orderBy) {
  if (b[orderBy] < a[orderBy]) {
    return -1
  }
  if (b[orderBy] > a[orderBy]) {
    return 1
  }
  return 0
}
/**
 * Function return a Function to sort elements
 * by orderBy
 * in the given order
 * @param {string} order 'desc/asc'
 * @param {string} orderBy '<string property of the sorting object>'
 */
export const getSorting = (order, orderBy) => {
  return order === 'desc' ? (a, b) => desc(a, b, orderBy) : (a, b) => -desc(a, b, orderBy)
}
