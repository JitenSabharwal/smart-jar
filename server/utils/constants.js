export const ALL_CONTAINER_ENDPOINT = '/api/v1/user'
export const ADD_CONTAINER_ENDPOINT = '/api/v1/user/addUser'
export const UPDATE_CONTAINER_ENDPOINT = '/api/v1/user/updateUser'
export const DELETE_CONTAINER_ENDPOINT = '/api/v1/user/removeUser'
export const DELETE_ALL_CONTAINER_ENDPOINT = '/api/v1/user/removeAllUsers'
