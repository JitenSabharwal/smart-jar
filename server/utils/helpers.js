const log = require('./logger/logger')
/**
 * Parse function to prepare output json
 * @param {object} obj {data: <object>, error:<boolean>, errorMsg: <String>}
 */
const parse = obj => {
  const result = {
    success: true,
  }
  if (obj.error) {
    result.success = false
    result.error = obj.error
  }
  result.data = obj.data
  return result
}

/**
 * Function creates output user object from the given data
 * @param {object} obj {
      itemName: String,
      quantity: String,
      timestamp: Date/String(DD/MM/YYYY),
    }
 */
const outputContainer = (arr) => {
  try {
    const containers = {
      itemName: arr[0].itemName,
      details: arr.map(a => ({quantity: a.quantity, timestamp: a.timestamp})).sort((a, b) => +new Date(b.timestamp) - +new Date(a.timestamp)),
    }
    return containers
  } catch (e) {
    log.error(e)
  }
  return {}
}

module.exports = {
  parse,
  outputContainer,
}
