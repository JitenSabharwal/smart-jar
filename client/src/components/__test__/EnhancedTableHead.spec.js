/* eslint-disable no-undef */
import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import EnhancedTableHead from '../EnhancedTableHead.jsx'
import Adapter from 'enzyme-adapter-react-16'
import { CONTAINER_TABLE_HEADS } from '../../helpers/constants'
const numSelected = 0
const selected = ''
const onRequestSort = () => {

}
const onSelectAllClick = () => {

}
const order = 'desc'
const orderBy = 'firstName'
const rowCount = 10
const rows = CONTAINER_TABLE_HEADS

Enzyme.configure({ adapter: new Adapter() })
describe('EnhancedTableHead', () => {
  test('render', () => {
    const wrapper = shallow(<EnhancedTableHead
      numSelected={numSelected}
      selected={selected}
      onRequestSort={onRequestSort}
      onSelectAllClick={onSelectAllClick}
      order={order}
      orderBy={orderBy}
      rowCount={rowCount}
      rows={rows}
    />)
    expect(wrapper.exists()).toBe(true)
    expect(wrapper).toMatchSnapshot()
  })
})
