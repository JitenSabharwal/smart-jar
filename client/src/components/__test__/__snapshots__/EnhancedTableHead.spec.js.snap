// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnhancedTableHead render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EnhancedTableHead
    numSelected={0}
    onRequestSort={[Function]}
    onSelectAllClick={[Function]}
    order="desc"
    orderBy="firstName"
    rowCount={10}
    rows={
      Array [
        Object {
          "disablePadding": true,
          "id": "firstName",
          "label": "FirstName",
          "numeric": false,
        },
        Object {
          "disablePadding": false,
          "id": "lastName",
          "label": "LastName",
          "numeric": false,
        },
        Object {
          "disablePadding": false,
          "id": "hobbies",
          "label": "Hobbies",
          "numeric": false,
        },
        Object {
          "disablePadding": false,
          "id": "birthDate",
          "label": "BirthDate",
          "numeric": false,
        },
        Object {
          "disablePadding": false,
          "id": "url",
          "label": "Profile",
          "numeric": false,
        },
      ]
    }
    selected=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(TableRow)>
        <WithStyles(TableCell)
          padding="checkbox"
        >
          <WithStyles(Checkbox)
            checked={false}
            indeterminate={false}
            onChange={[Function]}
          />
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          numeric={false}
          padding="none"
          sortDirection="desc"
        >
          <WithStyles(Tooltip)
            enterDelay={300}
            placement="bottom-start"
            title="Sort"
          >
            <WithStyles(TableSortLabel)
              active={true}
              direction="desc"
              onClick={[Function]}
            >
              FirstName
            </WithStyles(TableSortLabel)>
          </WithStyles(Tooltip)>
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          numeric={false}
          padding="default"
          sortDirection={false}
        >
          <WithStyles(Tooltip)
            enterDelay={300}
            placement="bottom-start"
            title="Sort"
          >
            <WithStyles(TableSortLabel)
              active={false}
              direction="desc"
              onClick={[Function]}
            >
              LastName
            </WithStyles(TableSortLabel)>
          </WithStyles(Tooltip)>
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          numeric={false}
          padding="default"
          sortDirection={false}
        >
          <WithStyles(Tooltip)
            enterDelay={300}
            placement="bottom-start"
            title="Sort"
          >
            <WithStyles(TableSortLabel)
              active={false}
              direction="desc"
              onClick={[Function]}
            >
              Hobbies
            </WithStyles(TableSortLabel)>
          </WithStyles(Tooltip)>
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          numeric={false}
          padding="default"
          sortDirection={false}
        >
          <WithStyles(Tooltip)
            enterDelay={300}
            placement="bottom-start"
            title="Sort"
          >
            <WithStyles(TableSortLabel)
              active={false}
              direction="desc"
              onClick={[Function]}
            >
              BirthDate
            </WithStyles(TableSortLabel)>
          </WithStyles(Tooltip)>
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          numeric={false}
          padding="default"
          sortDirection={false}
        >
          <WithStyles(Tooltip)
            enterDelay={300}
            placement="bottom-start"
            title="Sort"
          >
            <WithStyles(TableSortLabel)
              active={false}
              direction="desc"
              onClick={[Function]}
            >
              Profile
            </WithStyles(TableSortLabel)>
          </WithStyles(Tooltip)>
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(TableCell)
            padding="checkbox"
          >
            <WithStyles(Checkbox)
              checked={false}
              indeterminate={false}
              onChange={[Function]}
            />
          </WithStyles(TableCell)>,
          Array [
            <WithStyles(TableCell)
              numeric={false}
              padding="none"
              sortDirection="desc"
            >
              <WithStyles(Tooltip)
                enterDelay={300}
                placement="bottom-start"
                title="Sort"
              >
                <WithStyles(TableSortLabel)
                  active={true}
                  direction="desc"
                  onClick={[Function]}
                >
                  FirstName
                </WithStyles(TableSortLabel)>
              </WithStyles(Tooltip)>
            </WithStyles(TableCell)>,
            <WithStyles(TableCell)
              numeric={false}
              padding="default"
              sortDirection={false}
            >
              <WithStyles(Tooltip)
                enterDelay={300}
                placement="bottom-start"
                title="Sort"
              >
                <WithStyles(TableSortLabel)
                  active={false}
                  direction="desc"
                  onClick={[Function]}
                >
                  LastName
                </WithStyles(TableSortLabel)>
              </WithStyles(Tooltip)>
            </WithStyles(TableCell)>,
            <WithStyles(TableCell)
              numeric={false}
              padding="default"
              sortDirection={false}
            >
              <WithStyles(Tooltip)
                enterDelay={300}
                placement="bottom-start"
                title="Sort"
              >
                <WithStyles(TableSortLabel)
                  active={false}
                  direction="desc"
                  onClick={[Function]}
                >
                  Hobbies
                </WithStyles(TableSortLabel)>
              </WithStyles(Tooltip)>
            </WithStyles(TableCell)>,
            <WithStyles(TableCell)
              numeric={false}
              padding="default"
              sortDirection={false}
            >
              <WithStyles(Tooltip)
                enterDelay={300}
                placement="bottom-start"
                title="Sort"
              >
                <WithStyles(TableSortLabel)
                  active={false}
                  direction="desc"
                  onClick={[Function]}
                >
                  BirthDate
                </WithStyles(TableSortLabel)>
              </WithStyles(Tooltip)>
            </WithStyles(TableCell)>,
            <WithStyles(TableCell)
              numeric={false}
              padding="default"
              sortDirection={false}
            >
              <WithStyles(Tooltip)
                enterDelay={300}
                placement="bottom-start"
                title="Sort"
              >
                <WithStyles(TableSortLabel)
                  active={false}
                  direction="desc"
                  onClick={[Function]}
                >
                  Profile
                </WithStyles(TableSortLabel)>
              </WithStyles(Tooltip)>
            </WithStyles(TableCell)>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Checkbox)
              checked={false}
              indeterminate={false}
              onChange={[Function]}
            />,
            "padding": "checkbox",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checked": false,
              "indeterminate": false,
              "onChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "firstName",
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Tooltip)
              enterDelay={300}
              placement="bottom-start"
              title="Sort"
            >
              <WithStyles(TableSortLabel)
                active={true}
                direction="desc"
                onClick={[Function]}
              >
                FirstName
              </WithStyles(TableSortLabel)>
            </WithStyles(Tooltip)>,
            "numeric": false,
            "padding": "none",
            "sortDirection": "desc",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(TableSortLabel)
                active={true}
                direction="desc"
                onClick={[Function]}
              >
                FirstName
              </WithStyles(TableSortLabel)>,
              "enterDelay": 300,
              "placement": "bottom-start",
              "title": "Sort",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": true,
                "children": "FirstName",
                "direction": "desc",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "FirstName",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "lastName",
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Tooltip)
              enterDelay={300}
              placement="bottom-start"
              title="Sort"
            >
              <WithStyles(TableSortLabel)
                active={false}
                direction="desc"
                onClick={[Function]}
              >
                LastName
              </WithStyles(TableSortLabel)>
            </WithStyles(Tooltip)>,
            "numeric": false,
            "padding": "default",
            "sortDirection": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(TableSortLabel)
                active={false}
                direction="desc"
                onClick={[Function]}
              >
                LastName
              </WithStyles(TableSortLabel)>,
              "enterDelay": 300,
              "placement": "bottom-start",
              "title": "Sort",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "children": "LastName",
                "direction": "desc",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "LastName",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "hobbies",
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Tooltip)
              enterDelay={300}
              placement="bottom-start"
              title="Sort"
            >
              <WithStyles(TableSortLabel)
                active={false}
                direction="desc"
                onClick={[Function]}
              >
                Hobbies
              </WithStyles(TableSortLabel)>
            </WithStyles(Tooltip)>,
            "numeric": false,
            "padding": "default",
            "sortDirection": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(TableSortLabel)
                active={false}
                direction="desc"
                onClick={[Function]}
              >
                Hobbies
              </WithStyles(TableSortLabel)>,
              "enterDelay": 300,
              "placement": "bottom-start",
              "title": "Sort",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "children": "Hobbies",
                "direction": "desc",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Hobbies",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "birthDate",
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Tooltip)
              enterDelay={300}
              placement="bottom-start"
              title="Sort"
            >
              <WithStyles(TableSortLabel)
                active={false}
                direction="desc"
                onClick={[Function]}
              >
                BirthDate
              </WithStyles(TableSortLabel)>
            </WithStyles(Tooltip)>,
            "numeric": false,
            "padding": "default",
            "sortDirection": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(TableSortLabel)
                active={false}
                direction="desc"
                onClick={[Function]}
              >
                BirthDate
              </WithStyles(TableSortLabel)>,
              "enterDelay": 300,
              "placement": "bottom-start",
              "title": "Sort",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "children": "BirthDate",
                "direction": "desc",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "BirthDate",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "url",
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Tooltip)
              enterDelay={300}
              placement="bottom-start"
              title="Sort"
            >
              <WithStyles(TableSortLabel)
                active={false}
                direction="desc"
                onClick={[Function]}
              >
                Profile
              </WithStyles(TableSortLabel)>
            </WithStyles(Tooltip)>,
            "numeric": false,
            "padding": "default",
            "sortDirection": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(TableSortLabel)
                active={false}
                direction="desc"
                onClick={[Function]}
              >
                Profile
              </WithStyles(TableSortLabel)>,
              "enterDelay": 300,
              "placement": "bottom-start",
              "title": "Sort",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "children": "Profile",
                "direction": "desc",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Profile",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(TableRow)>
          <WithStyles(TableCell)
            padding="checkbox"
          >
            <WithStyles(Checkbox)
              checked={false}
              indeterminate={false}
              onChange={[Function]}
            />
          </WithStyles(TableCell)>
          <WithStyles(TableCell)
            numeric={false}
            padding="none"
            sortDirection="desc"
          >
            <WithStyles(Tooltip)
              enterDelay={300}
              placement="bottom-start"
              title="Sort"
            >
              <WithStyles(TableSortLabel)
                active={true}
                direction="desc"
                onClick={[Function]}
              >
                FirstName
              </WithStyles(TableSortLabel)>
            </WithStyles(Tooltip)>
          </WithStyles(TableCell)>
          <WithStyles(TableCell)
            numeric={false}
            padding="default"
            sortDirection={false}
          >
            <WithStyles(Tooltip)
              enterDelay={300}
              placement="bottom-start"
              title="Sort"
            >
              <WithStyles(TableSortLabel)
                active={false}
                direction="desc"
                onClick={[Function]}
              >
                LastName
              </WithStyles(TableSortLabel)>
            </WithStyles(Tooltip)>
          </WithStyles(TableCell)>
          <WithStyles(TableCell)
            numeric={false}
            padding="default"
            sortDirection={false}
          >
            <WithStyles(Tooltip)
              enterDelay={300}
              placement="bottom-start"
              title="Sort"
            >
              <WithStyles(TableSortLabel)
                active={false}
                direction="desc"
                onClick={[Function]}
              >
                Hobbies
              </WithStyles(TableSortLabel)>
            </WithStyles(Tooltip)>
          </WithStyles(TableCell)>
          <WithStyles(TableCell)
            numeric={false}
            padding="default"
            sortDirection={false}
          >
            <WithStyles(Tooltip)
              enterDelay={300}
              placement="bottom-start"
              title="Sort"
            >
              <WithStyles(TableSortLabel)
                active={false}
                direction="desc"
                onClick={[Function]}
              >
                BirthDate
              </WithStyles(TableSortLabel)>
            </WithStyles(Tooltip)>
          </WithStyles(TableCell)>
          <WithStyles(TableCell)
            numeric={false}
            padding="default"
            sortDirection={false}
          >
            <WithStyles(Tooltip)
              enterDelay={300}
              placement="bottom-start"
              title="Sort"
            >
              <WithStyles(TableSortLabel)
                active={false}
                direction="desc"
                onClick={[Function]}
              >
                Profile
              </WithStyles(TableSortLabel)>
            </WithStyles(Tooltip)>
          </WithStyles(TableCell)>
        </WithStyles(TableRow)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(TableCell)
              padding="checkbox"
            >
              <WithStyles(Checkbox)
                checked={false}
                indeterminate={false}
                onChange={[Function]}
              />
            </WithStyles(TableCell)>,
            Array [
              <WithStyles(TableCell)
                numeric={false}
                padding="none"
                sortDirection="desc"
              >
                <WithStyles(Tooltip)
                  enterDelay={300}
                  placement="bottom-start"
                  title="Sort"
                >
                  <WithStyles(TableSortLabel)
                    active={true}
                    direction="desc"
                    onClick={[Function]}
                  >
                    FirstName
                  </WithStyles(TableSortLabel)>
                </WithStyles(Tooltip)>
              </WithStyles(TableCell)>,
              <WithStyles(TableCell)
                numeric={false}
                padding="default"
                sortDirection={false}
              >
                <WithStyles(Tooltip)
                  enterDelay={300}
                  placement="bottom-start"
                  title="Sort"
                >
                  <WithStyles(TableSortLabel)
                    active={false}
                    direction="desc"
                    onClick={[Function]}
                  >
                    LastName
                  </WithStyles(TableSortLabel)>
                </WithStyles(Tooltip)>
              </WithStyles(TableCell)>,
              <WithStyles(TableCell)
                numeric={false}
                padding="default"
                sortDirection={false}
              >
                <WithStyles(Tooltip)
                  enterDelay={300}
                  placement="bottom-start"
                  title="Sort"
                >
                  <WithStyles(TableSortLabel)
                    active={false}
                    direction="desc"
                    onClick={[Function]}
                  >
                    Hobbies
                  </WithStyles(TableSortLabel)>
                </WithStyles(Tooltip)>
              </WithStyles(TableCell)>,
              <WithStyles(TableCell)
                numeric={false}
                padding="default"
                sortDirection={false}
              >
                <WithStyles(Tooltip)
                  enterDelay={300}
                  placement="bottom-start"
                  title="Sort"
                >
                  <WithStyles(TableSortLabel)
                    active={false}
                    direction="desc"
                    onClick={[Function]}
                  >
                    BirthDate
                  </WithStyles(TableSortLabel)>
                </WithStyles(Tooltip)>
              </WithStyles(TableCell)>,
              <WithStyles(TableCell)
                numeric={false}
                padding="default"
                sortDirection={false}
              >
                <WithStyles(Tooltip)
                  enterDelay={300}
                  placement="bottom-start"
                  title="Sort"
                >
                  <WithStyles(TableSortLabel)
                    active={false}
                    direction="desc"
                    onClick={[Function]}
                  >
                    Profile
                  </WithStyles(TableSortLabel)>
                </WithStyles(Tooltip)>
              </WithStyles(TableCell)>,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Checkbox)
                checked={false}
                indeterminate={false}
                onChange={[Function]}
              />,
              "padding": "checkbox",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "checked": false,
                "indeterminate": false,
                "onChange": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "firstName",
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Tooltip)
                enterDelay={300}
                placement="bottom-start"
                title="Sort"
              >
                <WithStyles(TableSortLabel)
                  active={true}
                  direction="desc"
                  onClick={[Function]}
                >
                  FirstName
                </WithStyles(TableSortLabel)>
              </WithStyles(Tooltip)>,
              "numeric": false,
              "padding": "none",
              "sortDirection": "desc",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(TableSortLabel)
                  active={true}
                  direction="desc"
                  onClick={[Function]}
                >
                  FirstName
                </WithStyles(TableSortLabel)>,
                "enterDelay": 300,
                "placement": "bottom-start",
                "title": "Sort",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": true,
                  "children": "FirstName",
                  "direction": "desc",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "FirstName",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "lastName",
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Tooltip)
                enterDelay={300}
                placement="bottom-start"
                title="Sort"
              >
                <WithStyles(TableSortLabel)
                  active={false}
                  direction="desc"
                  onClick={[Function]}
                >
                  LastName
                </WithStyles(TableSortLabel)>
              </WithStyles(Tooltip)>,
              "numeric": false,
              "padding": "default",
              "sortDirection": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(TableSortLabel)
                  active={false}
                  direction="desc"
                  onClick={[Function]}
                >
                  LastName
                </WithStyles(TableSortLabel)>,
                "enterDelay": 300,
                "placement": "bottom-start",
                "title": "Sort",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "children": "LastName",
                  "direction": "desc",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "LastName",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "hobbies",
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Tooltip)
                enterDelay={300}
                placement="bottom-start"
                title="Sort"
              >
                <WithStyles(TableSortLabel)
                  active={false}
                  direction="desc"
                  onClick={[Function]}
                >
                  Hobbies
                </WithStyles(TableSortLabel)>
              </WithStyles(Tooltip)>,
              "numeric": false,
              "padding": "default",
              "sortDirection": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(TableSortLabel)
                  active={false}
                  direction="desc"
                  onClick={[Function]}
                >
                  Hobbies
                </WithStyles(TableSortLabel)>,
                "enterDelay": 300,
                "placement": "bottom-start",
                "title": "Sort",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "children": "Hobbies",
                  "direction": "desc",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Hobbies",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "birthDate",
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Tooltip)
                enterDelay={300}
                placement="bottom-start"
                title="Sort"
              >
                <WithStyles(TableSortLabel)
                  active={false}
                  direction="desc"
                  onClick={[Function]}
                >
                  BirthDate
                </WithStyles(TableSortLabel)>
              </WithStyles(Tooltip)>,
              "numeric": false,
              "padding": "default",
              "sortDirection": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(TableSortLabel)
                  active={false}
                  direction="desc"
                  onClick={[Function]}
                >
                  BirthDate
                </WithStyles(TableSortLabel)>,
                "enterDelay": 300,
                "placement": "bottom-start",
                "title": "Sort",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "children": "BirthDate",
                  "direction": "desc",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "BirthDate",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "url",
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Tooltip)
                enterDelay={300}
                placement="bottom-start"
                title="Sort"
              >
                <WithStyles(TableSortLabel)
                  active={false}
                  direction="desc"
                  onClick={[Function]}
                >
                  Profile
                </WithStyles(TableSortLabel)>
              </WithStyles(Tooltip)>,
              "numeric": false,
              "padding": "default",
              "sortDirection": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(TableSortLabel)
                  active={false}
                  direction="desc"
                  onClick={[Function]}
                >
                  Profile
                </WithStyles(TableSortLabel)>,
                "enterDelay": 300,
                "placement": "bottom-start",
                "title": "Sort",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "children": "Profile",
                  "direction": "desc",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Profile",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
